scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -738,6 +751,25 @@ function Setup-Environment([string] $ContainerOs, [switch] $SkipArchCheck) {
             [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::Tls12
     }
 
+    if (($ContainerOs -eq 'Windows') -and (-not $SkipBatteryCheck)) {
+        if ((Get-Command 'Get-WmiObject' -ErrorAction Ignore) -ne $null) {
+            [psobject[]] $batteries = Get-WmiObject Win32_Battery -ErrorAction Ignore
+            if (($batteries -ne $null) -and ($batteries.Length -gt 0)) {
+                Write-Warning (
+                    "One or more batteries were detected on this device.`n`n" +
+                    'A known Windows operating system issue prevents transition to sleep and hibernate power states when IoT Edge modules ' +
+                    "(process-isolated Windows Nano Server containers) are running. This issue impacts battery life on the device.`n`n" +
```

Comment:
It would be valid for any Windows containers, right?

Reply:
Only nanoserver.

Reply:
I have not been following the battery thread closely. Are you sure this would not repro with IoTCore or ServerCore containers?

Reply:
ServerCore containers are definitely fine. For IoTCore containers - we don't use it for our own modules, so I'm not sure. You are probably in the best position to test :)